plugins/default/lib/mongodb_health.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+import argparse
+import sys
+import traceback
+from collections import Counter
+
+import openshift
+import nagios
+from mongodb import ReplsetStatuses
+
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', '--selector', type=str, action="store",
+                        required=True, help="Label Selector")
+    return parser
+
+
+check_mongodb_cmd = ("bash", "-c",
+                     "mongo 127.0.0.1/admin "
+                     "-u \"admin\" -p \"$MONGODB_ADMIN_PASSWORD\" "
+                     "--eval=\"rs.status().myState\" --quiet")
+
+
+def parse_mongo_result(output):
+    try:
+        return int(output.rstrip("\n"))
+    except:
+        return None
+
+
+def analize(status):
+    if status is None:
+        return nagios.UNKNOWN
+    elif status == ReplsetStatuses.PRIMARY or status == ReplsetStatuses.SECONDARY:
+        return nagios.OK
+    elif status == ReplsetStatuses.ARBITER or status == ReplsetStatuses.STARTUP:
+        return nagios.WARN
+    else:
+        return nagios.CRIT
+
+
+def report(pods, rs_statuses, nag_statuses):
+    if len(filter(None, pods)) < 1:
+        print "%s: Unable to locate any pods running mongodb" % (
+            nagios.status_code_to_label(nagios.UNKNOWN),)
+        return nagios.UNKNOWN
+
+    unique_rs_statuses = Counter(rs_status for rs_status in rs_statuses)
+    unique_nag_statuses = Counter(nag_status for nag_status in nag_statuses)
+
+    ret = max(unique_nag_statuses)
+
+    if unique_rs_statuses[ReplsetStatuses.PRIMARY] != 1:
+        ret = nagios.CRIT
+        print "%s: There are %s nodes claiming to have the primary replica set" % (
```

Comment:
If we are talking about testing things that never happen, we shouldn't have this check for multiple primaries, for it is not how MongoDB works. We are not testing that MongoDB have a broken implementation of replica sets :)